src/core/types.h

```diff
@@ -291,6 +293,7 @@ class SourceInfo {
     CFG* cfg;
     bool is_generator;
     std::string fn; // equivalent of code.co_filename
+    std::shared_ptr<LivenessAnalysis> liveness_info;
```

Comment:
I will change that one to a ```unique_ptr``` because we don't use ```shared_ptr``` at every use and having ```SourceInfo``` as the owner should be fine.